omod/src/main/webapp/fragments/field/datetimepicker.gsp

```diff
@@ -39,13 +45,29 @@
         startDate = defaultDateString
     } else {
         startDate = config.startDate
+        if (startDate instanceof String) {
+            try {
+                startDate = dateStringFormat.parse(startDate)
+            } catch(Exception e) {
+                dateStringFormat2 = new java.text.SimpleDateFormat("E MMM dd hh:mm:ss Z yyyy", org.openmrs.api.context.Context.getLocale())
```

Comment:
Where does this date format come from?  It seems pretty non-standard?  I'd be fine with supporting multiple formats, but unless this a common or standard format I'd rather support passing in a date string format?

Reply:
Well I know it's kind none standard, but we intend to use this velocity attribute: https://github.com/openmrs/openmrs-module-htmlformentry/blob/4fbef121d8192e21eb4df7058dd511a69fb0612b/api/src/main/java/org/openmrs/module/htmlformentry/FormEntrySession.java#L166

Reply:
@samuelmale is that really the format the velocity expression ends up in?

Let's add a comment explaining the reason then, otherwise people will never remember when they look at it again.

Reply:
Oohh sure!

Reply:
@mogoodrich does it look any better?

Reply:
@samuelmale can we move the comment into the try/catch block (ie before line 53)?

I think I understand this better now! I didn't realize that that format was the default returned by data.toString(), it's a really strange one in my opinion, but I guess it's  kind of a "standard" if it's what toString() returns...